File: ScaleformUI_Csharp/PauseMenu/Tabs/TextTab.cs
using CitizenFX.Core;
using System.Collections.Generic;
using System.Drawing;

namespace ScaleformUI.PauseMenu
{
    public class TextTab : BaseTab
    {
        public string TextTitle { get; set; }
        public int WordWrap { get; set; }
        public List<BasicTabItem> LabelsList = new List<BasicTabItem>();

        public TextTab(string name, string title) : base(name)
        {
            TextTitle = title;
            _type = 1;
        }

        public async void AddTitle(string title)
        {
            if(string.IsNullOrWhiteSpace(TextTitle))
                TextTitle = title;
        }

        public async void AddItem(BasicTabItem item)
        {
            LabelsList.Add(item);
        }
    }
}

File: ScaleformUI_Lua/src/PauseMenu/Tabs/TextTab.lua
TextTab = setmetatable({}, TextTab)
TextTab.__index = TextTab
TextTab.__call = function()
    return "BaseTab", "TextTab"
end

---@class TextTab

function TextTab.New(name, _title)
    local data = {
        Base = BaseTab.New(name or "", 0),
        Label = name,
        TextTitle = _title or "",
        LabelsList = {},
        LeftItemList = {},
        Index = 0,
        Focused = false,
        Parent = nil
    }
    return setmetatable(data, TextTab)
end

function TextTab:AddTitle(title)
    if not title:IsNullOrEmpty() then
        self.TextTitle = title
    end
end

function TextTab:AddItem(item)
    self.LabelsList[#self.LabelsList + 1] = item
end
